apps/frontend/src/libs/components/date-input/date-input.tsx

```diff
@@ -0,0 +1,104 @@
+import { Calendar } from "react-calendar";
+import {
+	type Control,
+	type FieldPath,
+	type FieldValues,
+} from "react-hook-form";
+
+import { Icon, Popover } from "~/libs/components/components.js";
+import { formatDate } from "~/libs/helpers/helpers.js";
+import {
+	useCallback,
+	useFormController,
+	usePopover,
+} from "~/libs/hooks/hooks.js";
+
+import { MILLIS_IN_DAY } from "./libs/constants/constants.js";
+import { getShortWeekday } from "./libs/helpers/helpers.js";
+import { type DateValue } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+type Properties<T extends FieldValues> = {
+	control: Control<T, null>;
+	maxDate?: Date;
+	maxRange?: number;
+	minDate?: Date;
+	name: FieldPath<T>;
+};
+
+const DateInput = <T extends FieldValues>({
+	control,
+	maxDate,
+	maxRange,
+	minDate,
+	name,
+}: Properties<T>): JSX.Element => {
+	const { field } = useFormController({
+		control,
+		name,
+	});
+
+	const { isOpened, onClose, onOpen } = usePopover();
+
+	const getSelectedDateRange = (): string => {
+		const [startDate, endDate] = field.value as Date[];
+
+		return `${formatDate(startDate as Date, "MMM d, yyyy")} - ${formatDate(endDate as Date, "MMM d, yyyy")}`;
+	};
+
+	const handleDateChange = useCallback(
+		(dates: DateValue): void => {
+			const [startDate, endDate] = dates as Date[];
+
+			if (maxRange && startDate && endDate) {
+				const diffInDays =
+					(endDate.getTime() - startDate.getTime()) / MILLIS_IN_DAY;
+
+				if (diffInDays > maxRange) {
+					const adjustedEndDate = new Date(startDate.getDate() + maxRange);
+					field.onChange([startDate, adjustedEndDate]);
+
+					return;
+				}
+			}
+
+			field.onChange(dates);
+		},
+		[field, maxRange],
+	);
+
+	return (
+		<div>
```

Comment:
We need to use <input /> tag  with type date for this component, and just override it's handlers

Reply:
I'm not quite sure, how to do it properly, but I tried, and there will be a lot of issues if we do it like this, for example we'll have only clickable icon, also when we click on input (even if it'll be readonly) it will get focus and characters there will become selected, and the main thing is that we will not be able to fill the input with our custom date output, because it'll throw an error. I understand that my approach is not as good for semantics, but it might be more convenient like this.

Reply:
Done. I'm not sure if this is what you meant, because there are not a lot of handlers to override if it is text😅. Also I'm not sure about leaving a button as a wrapper for this input and icon, but I think it will be more convenient for user to click on a whole element, and not only on that tiny calendar.